src/Rules/RegistryFactory.php

```diff
@@ -0,0 +1,31 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules;
+
+class RegistryFactory
+{
+
+	const RULE_TAG = 'phpstan.rules.rule';
+
+	/** @var \Nette\DI\Container */
+	private $container;
+
+	public function __construct(\Nette\DI\Container $container)
```

Comment:
This duplicates tag and DI logic and violates few SOLID principles. No need for that.

Rather use Extension and `beforeCompile()` method:

```php

class PhpStanExtension extends CompilerExtension
{

    public function beforeCompile()
    {
        $containerBuilder = $this->getContainerBuilder();

        $registryDefinition = $containerBuilder->getDefinition($containerBuilder->getByType(Registry::class);

        // or findByTag(), whatever you prefer        
        foreach ($containerBuilder->findByType(Rule::class) as $definitionName => $definition) {
		$registryDefinition->addSetup('register', ['@' . $definitionName]);
        }
    }
}
```